File: api-java-sql/src/main/java/com/nebulahotel/controller/HospedagensController.java
package com.nebulahotel.controller;

import com.nebulahotel.model.Hospedagens;
import com.nebulahotel.repository.HospedagensRepository;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@CrossOrigin
@RestController
@RequestMapping("/hospedagens")
public class HospedagensController {

    private final HospedagensRepository hospedagensRepository;

    public HospedagensController(HospedagensRepository hospedagensRepository) {
        this.hospedagensRepository = hospedagensRepository;
    }

    @GetMapping
    public List<Hospedagens> listarTodas() {
        return hospedagensRepository.findAll();
    }

    @GetMapping("/{id}")
    public ResponseEntity<Hospedagens> buscarPorId(@PathVariable Integer id) {
        return hospedagensRepository.findById(id)
                .map(ResponseEntity::ok)
                .orElseGet(() -> ResponseEntity.notFound().build());
    }

    @PostMapping
    public ResponseEntity<?> criar(@RequestBody Hospedagens hospedagem) {
        if (hospedagem.getIdHospedagem() == null) {
            return ResponseEntity.badRequest().body(Map.of("erro", "O campo idHospedagem é obrigatório."));
        }

        if (hospedagensRepository.existsById(hospedagem.getIdHospedagem())) {
            return ResponseEntity.status(HttpStatus.CONFLICT)
                    .body(Map.of("erro", "Já existe hospedagem com este idHospedagem."));
        }

        Hospedagens salvo = hospedagensRepository.save(hospedagem);
        return ResponseEntity.status(HttpStatus.CREATED).body(salvo);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Hospedagens> atualizar(@PathVariable Integer id, @RequestBody Hospedagens hospedagemAtualizada) {
        if (!hospedagensRepository.existsById(id)) {
            return ResponseEntity.notFound().build();
        }

        hospedagemAtualizada.setIdHospedagem(id);
        Hospedagens salvo = hospedagensRepository.save(hospedagemAtualizada);
        return ResponseEntity.ok(salvo);
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> remover(@PathVariable Integer id) {
        if (!hospedagensRepository.existsById(id)) {
            return ResponseEntity.notFound().build();
        }

        hospedagensRepository.deleteById(id);
        return ResponseEntity.noContent().build();
    }
}

File: api-kotlin-nosql/src/main/kotlin/com/nebulahotel/controller/HospedagensController.kt
package com.nebulahotel.controller

import com.nebulahotel.model.Hospedagens
import com.nebulahotel.repository.HospedagensRepository
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@CrossOrigin
@RestController
@RequestMapping("/hospedagens")
class HospedagensController(
    private val hospedagensRepository: HospedagensRepository
) {

    @GetMapping
    fun listarTodas(): List<Hospedagens> = hospedagensRepository.findAll()

    @GetMapping("/{id}")
    fun buscarPorId(@PathVariable id: Int): ResponseEntity<Hospedagens> {
        val hospedagem = hospedagensRepository.findById(id)
        return if (hospedagem.isPresent) ResponseEntity.ok(hospedagem.get()) else ResponseEntity.notFound().build<Hospedagens>()
    }

    @PostMapping
    fun criar(@RequestBody hospedagem: Hospedagens): ResponseEntity<Any> {
        if (hospedagensRepository.existsById(hospedagem.idHospedagem)) {
            return ResponseEntity.status(HttpStatus.CONFLICT)
                .body(mapOf("erro" to "Já existe hospedagem com este idHospedagem."))
        }
        return ResponseEntity.status(HttpStatus.CREATED).body(hospedagensRepository.save(hospedagem))
    }

    @PutMapping("/{id}")
    fun atualizar(@PathVariable id: Int, @RequestBody hospedagemAtualizada: Hospedagens): ResponseEntity<Hospedagens> {
        if (!hospedagensRepository.existsById(id)) {
            return ResponseEntity.notFound().build<Hospedagens>()
        }
        val salvo = hospedagensRepository.save(hospedagemAtualizada.copy(idHospedagem = id))
        return ResponseEntity.ok(salvo)
    }

    @DeleteMapping("/{id}")
    fun remover(@PathVariable id: Int): ResponseEntity<Void> {
        if (!hospedagensRepository.existsById(id)) {
            return ResponseEntity.notFound().build()
        }
        hospedagensRepository.deleteById(id)
        return ResponseEntity.noContent().build()
    }
}
